### ISSUE TITLE
--display error in CompletionServer

### ISSUE THREAD

**Issue by R32** [HaxeFoundation/haxe#6916]
[haxedevelop proj.zip](https://github.com/HaxeFoundation/haxe/files/1836433/js.zip)
Main.hx
```haxe
@:build(Macros.build())
class Main {
	static function main() {
		trace("hello owrld!");
		trace(Main);
	}
}
```
Macros.hx
```haxe
class Macros {
	public static function build() {
		var t = haxe.macro.Context.getType("js.html.DOMElement");
		switch(t) {
		case TInst(r, _):
			var c = r.get();
			trace(r.toString()+ "  ---  " +c.superClass.t.toString());
		default:
		}
		return null;
	}
}
```
As shown in GIF, the **`.superClass.t.toString`** should return a full name: `js.html.Node` 

![error](https://user-images.githubusercontent.com/6825113/37755910-490b0b38-2de2-11e8-83cc-332192ba80b1.gif)

**Comment by R32** [comment]
I'm not sure, maybe it's just caused by haxedevelop.

my haxe version is `git@6b91eab23` and haxedevelop is `5.3.3.1@master#436add6fc7`

**Comment by ncannasse** [comment]
Try using -D old-error-position ?

**Comment by R32** [comment]
I have tried to add `-D old-error-format`, but still the same. I can't reproduce it from the command line, so it's probably caused by haxedevelop

**Comment by R32** [comment]
I found it started with haxe 3.4.0-rc.1, In the case of  `haxedevelop + CompletionServer`, the property **`classType.superClass.get().pack`** will be empty. 

fortunately, in many cases that can use `.superClass.get().module` to bypass it. 

So I guess some Completion fails issue like #6740 may be related to it

**Comment by Simn** [comment]
I cannot reproduce this, it always traces `js.html.Node` for me. So I'll assume that this has been addressed or requires very particular circumstances to reproduce.

**Comment by R32** [comment]
Finally, I feel that this issue has been solved by 602fddb06e9e06c2969504469f6f189cb2bf7928
